spec/factories/tailoring.rb

```diff
@@ -41,7 +41,7 @@
 
           result = sec_guide_rules + profile_rules
           # Make sure that the randomized result doesn't match the profile tailoring
-          break if result.to_set(&:id) != profile.rules.to_set(&:id)
+          break if result.to_set(&:id) != profile.rules.to_set(&:id) || profile.rules == []
```

Comment:
This is a bit reversed logic, `with_mixed_rules` should fail if there are no rules to mix...